File: extensions/problem/test/kotlin/ProblemTest.kt
import org.hoshino9.luogu.problem.*
import org.hoshino9.luogu.test.BaseTest
import org.hoshino9.luogu.test.printAllMember
import org.junit.Test

class ProblemTest : BaseTest() {
	@Test
	fun problemList() {
		client.problemList().printAllMember()
	}

	@Test
	fun problemInfo() {
		ProblemPageBuilder("P1000", client).build().problem.printAllMember()
	}

	@Test
	fun solutions() {
		SolutionPageBuilder("P1000", client).build().printAllMember()
	}

//	@Test
//	fun mark() {
//		runBlocking {
//			luogu.unmark("P3695")
//		}
//	}
}

File: src/test/scala/org/hoshino9/luogu/ProblemTest.scala
package org.hoshino9.luogu

import org.hoshino9.luogu.problem.{Problem, ProblemBase}
import org.junit.Test
import play.api.libs.json.Json
import problem.ProblemList._

class ProblemTest {
	@Test
	def testList(): Unit = {
		val client = LuoGuClient()
		println(client.problems().get)
	}

	@Test
	def testDetail(): Unit = {
		implicit val client: LuoGuClient = LuoGuClient()
		println(client.problems().map(_.result.head.lift.get).get)
	}
}
